File: java/Fase3.java
import java.util.ArrayList;
import java.util.Scanner;

public class Fase3 {
  public static void main(String[] args) { 
	  
	  int b5=5, b10=10, b20=20, b50=50, b100=100, b200=200, b500=500, totalb=0;
	  ArrayList<String> dish = new ArrayList<String>();
	  ArrayList<Integer> price = new ArrayList<Integer>();
	  Scanner input= new Scanner(System.in);

	  // crear menu
	  System.out.println("Crear Menu ");
	  System.out.println("Introduce nombre de los platos: ");
	  int exceptionChecker = 0;
	  
	  try{
		  for (int i=0; i<3;i++) {	  
			  System.out.println("Plato " + (i+1) + ": " );
			  dish.add(input.nextLine());	
		  }		  
		  System.out.println("Introduce precio: ");
		  
		  for (int j=0; j<dish.size();j++) {
			  System.out.println(dish.get(j) + ":");
			  price.add(input.nextInt());	
		  }		  
	  } catch (Exception e) {
		  System.out.println("Dato introducido no valido. Precio debe ser un numero entero.");	
		  exceptionChecker = 1;
	  }
		  
	  //imprimir menu
	  if(exceptionChecker==0) {
		  System.out.println("\t \t MENU ");
		  System.out.println("  PLATO \t \t Precio ");
	  
		  for (int i=0; i<dish.size();i++) {
		  System.out.print((i+1) +". " + dish.get(i) + "\t\t\t\t\t " + price.get(i) +"\n");
		  }
	  
	  //order 
		  ArrayList<Integer> order = new ArrayList <Integer>();
		  System.out.println("\nElige platos" );
		  System.out.println("Digite 0(zero) para finalizar" );
	
		  try {
			  int option=1;
			  while( option!=0) {
				  option=input.nextInt();
				  if (option==0) {
					  break;
				  }
				  order.add(option);	
			  }
			  input.close();
			  
		  //fase3
			  System.out.println("\n\nMenu elegido: ");
		 
			  for(int k=0; k< order.size(); k++) {
				  if(order.get(k)<=dish.size()) {
					  System.out.println(order.get(k) +". "  + dish.get(order.get(k)-1) + "\t\t\t\t\t " + price.get(order.get(k)-1));
					  totalb+=price.get(order.get(k)-1);
				  }else {
					  System.out.println("Plato elegido no existe");
					  totalb=0;
					  break;
				  }			  
			  }
			  
			  if(totalb!=0) {
				  System.out.println("Total " + "\t\t\t\t\t "  + totalb);
			  }
		   } catch (Exception e){
				  System.out.println("Dato introducido no valido.");
		   }
	  }
  }
}

File: javaScript/Fase3.js

	function funcionFase3(){

	const b5=5, b10=10, b20=20, b50=50, b100=100, b200=200, b500=500;
	var totalb=0, dish=[""], price=[0];
	
	//Crear Menu
	alert("Crear Menu");
	for (var i=0;i<3;i++){
		dish[i]=prompt('Ingrese el plato'+(i+1) + ":"); 
		price[i]=prompt('Ingrese el precio de '+dish[i]+ ":"); 
		
		while(isNaN(price[i])){// "NaN" cuando se presiona barra espaciadora o no se introduce dato
			price[i]=prompt("El precio debe ser un numero entero.\n Ingrese el precio de " +dish[i] + ":")
		}
	}  	
	
	//Mostrar Menu
	
	var menu="";
	for (var i=0;i<dish.length;i++){
			menu+= dish[i]  + " --------------------------------- "+ price[i] + "<br>";
	}
	document.getElementById('show1').innerHTML = "<b>Menu</b> <br><br>" + menu;
	
	
	//Elegir platos	
	
	var count=0, nextOrder=1, order=[""];
	alert("Inciar pedido");
	while(nextOrder==1){
		order[count]=prompt("Elige plato");
		nextOrder=parseInt(prompt("Desea continuar? 1:Si / 0:No"));
		while( nextOrder !=1 && nextOrder!=0){
			nextOrder=parseInt(prompt("Opcion no valida. \n Digite 1 (para continuar), 0 (para finalizar pedido)"));
		}
		count++;
	}
	
	//fase3
	var searchElement=0, printOrder="";
	for(var i=0; i<count;i++){
		searchElement= dish.indexOf(order[i]);
			if(searchElement==-1){
			document.getElementById("show2").innerHTML = "Producto elegido (" +order[i]+ ") no existe.";
			break;
			}else{
			printOrder+= order[i]+ " --------------------------------- " + price[searchElement]+ "<br>";
			totalb+=parseInt(price[searchElement]);
			document.getElementById('show2').innerHTML = "<b>Platos elegidos</b><br><br>"+printOrder+ "<br>" +
														 "Total --------------------------------- "+ totalb+ "<br>";
			}
	}
}	
